packages/massa-web3/src/experimental/client/publicAPI.ts

```diff
@@ -78,11 +85,48 @@ export class PublicAPI {
   }
 
   async executeReadOnlyCall(
-    readOnlyCall: ReadOnlyCall
-  ): Promise<ExecuteReadOnlyResponse> {
-    return this.connector
-      .execute_read_only_call([readOnlyCall])
-      .then((r) => r[FIRST])
+    readOnlyCall: ReadOnlyCallParams
+  ): Promise<ReadOnlyCallResult> {
+    const [readOnlyRes] = await this.connector.execute_read_only_call([
+      {
+        max_gas: Number(readOnlyCall.maxGas),
+        target_address: readOnlyCall.targetAddress,
+        target_function: readOnlyCall.func,
+        parameter: readOnlyCall.parameter
+          ? Array.from(readOnlyCall.parameter)
+          : [],
+        caller_address: readOnlyCall.callerAddress,
+        coins: readOnlyCall.coins
+          ? Mas.toString(readOnlyCall.coins).padStart(Mas.NB_DECIMALS, '0')
```

Comment:
Separation of concerns: it's not the job of the caller to handle padding, it should be done at the Mas level if needed.